### PR TITLE
[STY] Explicit exception raised in the assert message

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Raising a built-in or a custom exception when `assert` fails is ineffective. Asserts always raise an `AssertionError`, making the provided exception which the user is expecting useless.

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: nilearn/plotting/cm.py [hunk 1]
```diff
@@ -344,13 +344,9 @@ def dimmer(c):
...
     """Replace a colormap by another inside a pair of values."""
-    assert vmin < vmax, ValueError("vmin must be smaller than vmax")
-    assert vmin >= 0, ValueError(
-        f"vmin must be larger than 0, {vmin} was passed."
-    )
-    assert vmax <= 1, ValueError(
-        f"vmax must be smaller than 1, {vmax} was passed."
-    )
+    assert vmin < vmax, "vmin must be smaller than vmax"
+    assert vmin >= 0, f"vmin must be larger than 0, {vmin} was passed."
```
Comment (Remi-Gau) [inline nilearn/plotting/cm.py:348]:
```suggestion
    assert vmin >= 0, f"vmin must be larger than 0, {vmin=} was passed."
```

File: nilearn/plotting/cm.py [hunk 2]
```diff
@@ -344,13 +344,9 @@ def dimmer(c):
...
+    assert vmax <= 1, f"vmax must be smaller than 1, {vmax} was passed."
```
Comment (Remi-Gau) [inline nilearn/plotting/cm.py:349]:
```suggestion
    assert vmax <= 1, f"vmax must be smaller than 1, {vmax=} was passed."
```

File: nilearn/plotting/cm.py [hunk 3]
```diff
@@ -344,13 +344,9 @@ def dimmer(c):
...
 def replace_inside(outer_cmap, inner_cmap, vmin, vmax):
     """Replace a colormap by another inside a pair of values."""
-    assert vmin < vmax, ValueError("vmin must be smaller than vmax")
-    assert vmin >= 0, ValueError(
-        f"vmin must be larger than 0, {vmin} was passed."
-    )
-    assert vmax <= 1, ValueError(
-        f"vmax must be smaller than 1, {vmax} was passed."
-    )
+    assert vmin < vmax, "vmin must be smaller than vmax"
```
Comment (Remi-Gau) [inline nilearn/plotting/cm.py:347]:
```suggestion
    assert vmin < vmax, f"vmin must be smaller than vmax. Got {vmin=} and {vmax=}."
```

**Review by Remi-Gau — COMMENTED** [review]
Added some suggestions that should make the error message a bit more readable.
WDYT?

**Review by Remi-Gau — APPROVED** [review]
thanks

**Pull request merged**